ratis-examples/src/main/java/org/apache/ratis/examples/filestore/cli/DataStream.java

```diff
@@ -0,0 +1,145 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.ratis.examples.filestore.cli;
+
+import com.beust.jcommander.Parameter;
+import com.beust.jcommander.Parameters;
+import org.apache.ratis.client.RaftClient;
+import org.apache.ratis.client.api.DataStreamOutput;
+import org.apache.ratis.examples.filestore.FileStoreClient;
+import org.apache.ratis.protocol.DataStreamReply;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.nio.MappedByteBuffer;
+import java.nio.channels.FileChannel;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+
+/**
+ * Subcommand to generate load in filestore data stream state machine.
+ */
+@Parameters(commandDescription = "Load Generator for FileStore DataStream")
+public class DataStream extends Client {
+
+  @Parameter(names = {"--type"}, description = "DirectByteBuffer, MappedByteBuffer, transferTo", required = true)
+  private String dataStreamType;
+
+  @Override
+  protected void operation(RaftClient client) throws IOException {
+    List<String> paths = generateFiles();
+    FileStoreClient fileStoreClient = new FileStoreClient(client);
+    System.out.println("Starting DataStream write now ");
+
+    long startTime = System.currentTimeMillis();
+
+    long totalWrittenBytes = waitStreamFinish(streamWrite(paths, fileStoreClient));
+
+    long endTime = System.currentTimeMillis();
+
+    System.out.println("Total files written: " + numFiles);
+    System.out.println("Each files size: " + fileSizeInBytes);
+    System.out.println("Total data written: " + totalWrittenBytes + " bytes");
+    System.out.println("Total time taken: " + (endTime - startTime) + " millis");
+
+    client.close();
+    System.exit(0);
+  }
+
+  private Map<String, List<CompletableFuture<DataStreamReply>>> streamWrite(
+      List<String> paths, FileStoreClient fileStoreClient) throws IOException {
+    Map<String, List<CompletableFuture<DataStreamReply>>> fileMap = new HashMap<>();
+    for(String path : paths) {
+      File file = new File(path);
+      FileInputStream fis = new FileInputStream(file);
+      final DataStreamOutput dataStreamOutput = fileStoreClient.getStreamOutput(path, (int) file.length());
+
+      if (dataStreamType.equals("DirectByteBuffer")) {
+        fileMap.put(path, writeByDirectByteBuffer(dataStreamOutput, fis.getChannel()));
+      } else if (dataStreamType.equals("MappedByteBuffer")) {
+        fileMap.put(path, writeByMappedByteBuffer(dataStreamOutput, fis.getChannel()));
+      } else if (dataStreamType.equals("transferTo")) {
```

Comment:
Let's call this "FilePositionCount" or "NettyFileRegion".  TransferTo sounds like FileChannel.transferTo.